### PR TITLE
Build clang-included runtime, et al with associated top-level targets.

### PR DESCRIPTION
**Pull request by thomasvandoren**
Previously, the clang-included runtime and speculative third party builds were
a separate target, llvm-runtime-if-needed, which was invoked from comprt. This
worked well when the default target or the comprt target was called. However,
it was less than ideal when runtime or third-party-try-opt was invoked
directly.

For example, if a developer was working on the runtime and iteratively calling
`make runtime` from the top level, the clang-included runtime was not rebuilt
with each rebuild.

Update the makefile to build the clang-included runtime as a part of each of
the runtime and third-party-try-{gmp,re2} targets. Now, when the runtime is
built, the clang-included runtime will be built. The same is true for the
speculative gmp and re2 builds.

This will fix the issue with llvm and gasnet.llvm nightly testing configs. ./nightly
does not call `make` at the top-level, instead it calls the individual targets
in comprt (this is done for better failure reporting as well as to retry some
when they fail). It was not updated to call llvm-runtime-if-needed.

### Verification:

* [x] Call `make runtime` and ensure gen/ directories have regular and clang-included variations.
* [x] Call `make third-party-try-opt` and ensure gmp/install/ and re2/install/ dirs have clang-included variations.
* [x] Verify nightly -cron -hello  passes with llvm enabled.

paired with @bradcray on initial implementation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -85,11 +81,19 @@ third-party-try-opt: third-party-try-re2 third-party-try-gmp
...
+	if [ "llvm" = `${CHPL_MAKE_HOME}/util/chplenv/chpl_llvm.py` ]; then \
+	export CHPL_TARGET_COMPILER=clang-included && \
+	$(MAKE) try-re2; \
+	fi \
 	fi
 
 third-party-try-gmp: FORCE
 	-@if [ -z "$$CHPL_GMP" ]; then \
 	cd third-party && $(MAKE) try-gmp; \
+	if [ "llvm" = `${CHPL_MAKE_HOME}/util/chplenv/chpl_llvm.py` ]; then \
```
Comment (ronawho) [inline Makefile]: Should these be `if [ "none" != `${CHPL_MAKE_HOME}/util/chplenv/chpl_llvm.py` ]` ?

File: Makefile [hunk 2]
```diff
@@ -68,13 +67,10 @@ modules: FORCE
 
 runtime: FORCE
 	cd runtime && $(MAKE)
-
-llvm-runtime-if-needed: FORCE
 	-@if [ "llvm" = `${CHPL_MAKE_HOME}/util/chplenv/chpl_llvm.py` ]; then \
-	echo "Building runtime for chpl --llvm"; \
+	echo "Building runtime for chpl --llvm" ; \
```
Comment (ronawho) [inline Makefile]: Why do we echo this for llvm? Was this some debug that can/should be removed? If not, should it be added to re2 and gmp?

File: Makefile (hunk 1 above)
Comment (thomasvandoren) [inline Makefile]: Hmm, that's an interesting question. Right now clang-included means use the clang in third-party/llvm/install/... If we supported a system llvm, e.g. CHPL_LLVM=system, I'm not sure what would mean w.r.t. the target compiler. Let's discuss with Michael outside this PR.

Comment (mppf) [inline Makefile]: I think we'll have to figure it out if/when we support CHPL_LLVM=system. I *think* that the
answer is that in that case you'd need to set CHPL_TARGET_COMPILER=clang
even if it was originally GCC. But that's a bunch of other changes I don't think we can
predict right now.

Comment (thomasvandoren) [inline Makefile]: Ok, that sounds good. Thanks, Michael!

**Pull request merged**